### PR TITLE
Merge stored targets into one `StoredTargetsMap`

### PR DESCRIPTION
**Pull request by pokey**
## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-engine/src/core/StoredTargets.ts [hunk 1]
```diff
@@ -1,10 +1,25 @@
+import { DefaultMap } from "@cursorless/common";
 import { Target } from "../typings/target.types";
 
 /**
  * Used to store targets between commands.  This is used by marks like `that`
  * and `source`.
  */
-export class StoredTargets {
+export class StoredTargetMap {
+  private targetMap: DefaultMap<string, StoredTargets> = new DefaultMap(
```
Comment (AndreasArvidsson) [inline packages/cursorless-engine/src/core/StoredTargets.ts:9]: string? STRING?? I assume that is for future bookmarks, but is there anyway we could type our known stored targets?

**Review by AndreasArvidsson — APPROVED** [review]

File: packages/cursorless-engine/src/core/StoredTargets.ts (hunk 1 above)
Comment (pokey) [reply packages/cursorless-engine/src/core/StoredTargets.ts:9]: Haha yeah that's for custom bookmarks. I could just add separate fields for `that` and `source` and then the string getter would be just for custom use?

Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/core/StoredTargets.ts:9]: Sounds good!

Comment (pokey) [reply packages/cursorless-engine/src/core/StoredTargets.ts]: Ok for now I just made it strongly typed.  We can handle custom bookmarks when we get there

**Pull request merged**